### PR TITLE
build(deps): use upstream str0m

### PR DESCRIPTION
**Pull request by thomaseizinger**
With the adoption of #5080, connlib is now resilient against temporarily failed connections as they'll be immediately re-established. Thus, we no longer need any of the patches that we are currently maintaining in our str0m fork.

The only difference is an adjustment of the ICE timeout parameters but those can be made configurable in str0m.

Related: https://github.com/algesten/str0m/pull/537.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
So the build failure here surfaces and interesting point: At the moment, we have a patch in str0m in our fork that ensures we don't instantly flap to `Disconnected` in case we invalidate all candidates but already have new, untested ones. Originally, I added this exactly for the "migrate relay" usecase. However, I do think it is flawed.

When the portal tells us to discontinue use of a relay and gives us new ones, we instantly remove those relays and invalidate their candidates. If a connection has that relay has its nominated candidate, invalidating it will instantly attempt to use another candidate pair which can only be another relay because we wouldn't have picked a relay if hole-punching would have been available.

At the moment, we use all relays for all connections when are always given at least 2, meaning we should be able to migrate to the 2nd one in case the first one dies.

I've tried to rewrite this test to cover that but it is trickier than I thought because the snownet test harness doesn't support dispatching traffic between two relays. I don't really want to invest in that test harness, I'd much rather delete that code and cover it using `tunnel_test`.

Meaning, before we can merge this, I'll see if I can change the `tunnel_test` harness to do a minimal version of proper routing. Once that is in place, we can extend the `tunnel_test` harness with multiple relays (and later multiple gateways) and then delete this test without losing any confidence.

**Comment by thomaseizinger** [comment]
We can pick this back up once https://github.com/firezone/firezone/pull/6060 merges.

**Comment by thomaseizinger** [comment]
@conectado Ready for review.

File: rust/connlib/snownet/src/node.rs
Comment (thomaseizinger) [inline rust/connlib/snownet/src/node.rs:1821]: These are the settings we also currently use in our fork. See first commit in https://github.com/algesten/str0m/compare/main...firezone:str0m:main.

We may want to tweak these eventually but for the purpose of this PR, I'd like to keep them as is and only focus on the dependency upgrade.

**Review by conectado — APPROVED** [review]

**Pull request merged**
